api/transfer.py

```diff
@@ -132,10 +137,26 @@ async def post(self, request) -> JSONResponse:
         request_body = await request.json()
         user_id = request_body["user_id"]
         auth_manager.check_project_access_from_user_id(user_id, project_id)
+
+        records = request_body["records"]
+
+        project = project_manager.get_project(project_id)
+        num_project_records = len(project.records)
+        for attribute in project.attributes:
+            if attribute.is_primary_key:
+                for idx, record in enumerate(records):
+                    if attribute.name not in record:
+                        if attribute.name == "running_id":
+                            records[idx][attribute.name] = num_project_records + idx + 1
+                        else:
+                            raise exceptions.InvalidInputException(
+                                f"Non-running-id, primary key {attribute.name} missing in record"
+                            )
```

Comment:
Are missing fields allowed when they are not primary keys? I think we forbid it for the "normal" upload but maybe this is a wrong information on my side

Reply:
They should be allowed, just considered as NaN fields. I think this is already happening in refinery, right?